File: spotify-clone-backend/src/models/Song.js
const mongoose = require("mongoose");

const songSchema = new mongoose.Schema(
  {
    title: {
      type: String,
      required: true,
      trim: true
    },
    artistName: {
      type: String,
      required: true,
      trim: true
    },
    artist: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Artist",
      default: null
    },
    albumName: {
      type: String,
      default: "",
      trim: true
    },
    album: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "Album",
      default: null
    },
    genre: {
      type: String,
      default: "",
      trim: true
    },
    durationSeconds: {
      type: Number,
      default: 0
    },
    coverUrl: {
      type: String,
      default: ""
    },
    audioUrl: {
      type: String,
      required: true
    },
    createdBy: {
      type: mongoose.Schema.Types.ObjectId,
      ref: "User",
      default: null
    }
  },
  {
    timestamps: true
  }
);

module.exports = mongoose.model("Song", songSchema);

File: SpotifyClone/app/src/main/java/com/example/spotifyclone/data/model/Song.kt
package com.example.spotifyclone.data.model

data class Song(
    val _id: String,
    val title: String,
    val artistName: String? = null,
    val albumName: String? = null,
    val genre: String? = null,
    val durationSeconds: Int? = null,
    val coverUrl: String? = null,
    val audioUrl: String,
    val artist: ArtistRef? = null,
    val album: AlbumRef? = null
)

data class ArtistRef(
    val _id: String,
    val name: String
)

data class AlbumRef(
    val _id: String,
    val title: String
)